docs/developers/napari_architecture.md

```diff
@@ -0,0 +1,164 @@
+    @property
+    def temperature(self):
+        """Degrees in Fahrenheit."""
+        return self._temperature
+
+    @temperature.setter
+    def temperature(self, value):
+        self._temperature = value
+        # emit the temperature "changed" event
+        self.events.temperature(value=value)
+
+    @property
+    def humidity(self):
+        """Percent humidity."""
+        return self._humidity
+
+    @humidity.setter
+    def humidity(self, value):
+        self._humidity = value
+        # emit the humidity "changed" event
+        self.events.humidity(value=value)
+
+    @property
+    def wind(self):
+        """Wind speed in mph."""
+        return self._wind
+
+    @wind.setter
+    def wind(self, value):
+        self._wind = value
+        # emit the wind "changed" event
+        self.events.wind(value=value)
+```
+
+Another object can then "listen" for changes in our weather model and register
+a callback function with the event emitter of the attribute they would like
+to watch:
+
+```python
+# create an instance of the model
+weather = Weather(temperature=72, humidity=65, wind=0)
+
+# define some callback that should respond to changes in the model
+def hurricane_watch(event):
+    if event.value > 74:
+        print("Hurricane! Evacuate!")
+
+# register our callback with the model
+weather.events.wind.connect(hurricane_watch)
+
+# now, everytime weather.wind is changed, hurricane_watch is called
+weather.wind = 90  # prints: "Hurricane! Evacuate!"
+```
+
+This method is very customizable but requires a lot of boilerplate. The
+generic base model `EventedModel` was added to reduce this and
+"standardize" this change/emit pattern. The `EventedModel` provides the
+following features:
+
+* type validation and coersion on class instantiation and attribute assignment
+* event emission after successful attribute assignment
+
+Using `EventedModel` would reduce the above `weather` class code to:
+
+```python
+class weather(EventedModel):
+    """A simple model to track changes in the weather.
+
+    Parameters
+    ----------
+    temperature : float
+        Degrees in Fahrenheit.
+    humidity : float
+        Percent humidity
+    wind : float
+    """
+    temperature: float
+    humidity: float
+    wind: float
+```
+
+Currently most of the models in `napari/components/` are `EventedModels` but
+not the layer models although there is intention to convert these to
+`EventedModels` in the future.
+
+## Qt models
+
+Qt classes are responsible for all napari's user interface elements. There is
+generally one to one mapping between Python models and qt models in napari, for
+example Python model `Dims` and Qt model `QtDims`.
+The Qt class can register callbacks such that when an attribute of the
+corresponding Python model changes, the appropriate actions are taken.
+The Qt classes are also instantiated with a reference to
+the Python model, which gets updated directly when a field is changed via the
+GUI.
+
+## Vispy models
```

Comment:
Again, I'd avoid using the world *models* for similar reasons as the Qt section. I think classes or *Vispy layers* would be fine.

If I understand napari's architecture correctly, the vispy classes represent a view on the layers (which represent a part of the core model/data of the app) - like you mentioned that specifically updates the canvas (which in the case of napari right now does have a Qt widget backing it). In some sense the vispy classes also represent a model for data that will be sent to the GPU, but that's probably a bit nuanced to include here.